include/zephyr/bluetooth/mesh/access.h

```diff
@@ -145,19 +145,23 @@ extern "C" {
  *  @param _mods      Array of models.
  *  @param _vnd_mods  Array of vendor models.
  */
-#define BT_MESH_ELEM(_loc, _mods, _vnd_mods)        \
-{                                                   \
-	.loc              = (_loc),                 \
-	.model_count      = ARRAY_SIZE(_mods),      \
-	.vnd_model_count  = ARRAY_SIZE(_vnd_mods),  \
-	.models           = (_mods),                \
-	.vnd_models       = (_vnd_mods),            \
+#define BT_MESH_ELEM(_loc, _mods, _vnd_mods)		\
+{							\
+	.rt		  = (void *)(void *[]) { NULL },\
```

Comment:
I really do not like this magic. What is the size of unnamed structure? Why do you allocate `pointer size` memory for 2 bytes field? 
It should be like this to be precise with operated types:
```
.rt        = &(struct runtime_ctx) { .addr = 0 },\
```

Reply:
The same code has been approved by you in the previous PR: https://github.com/zephyrproject-rtos/zephyr/pull/65152
Should we change this for bt_mesh_model then?

Reply:
Done.